woocommerce/woocommerce-ios
# [Mobile Payments] Upsell Card Reader Feature announcement analytics

Closes: #7230 


### Description


Adds the following analytics events:

- `woocommerceios_feature_card_shown`
- `woocommerceios_feature_card_dismissed`
- `woocommerceios_feature_card_cta_tapped`

with these properties:

- `source`. In this case, the value will be `payment_methods`, and in future `order_list`, or `settings`
- `campaign_name`. In this case, the value will be `upsell_card_readers`

N.B. This includes a refactor of the screen the analytics are for, to rely more on the new `FeatureAnnouncementCardViewModel` for configuration of both the view, and the analytics logged.

### Testing instructions


Using a store based in the US or Canada:

1. Launch the app and go to the Orders tab
2. Tap `+` and select `Simple Payment` or `Create Order
3. Set up the payment/order and go through to the payment method selection screen.
4. Observe that the `shown` event is logged with the correct properties
5. Tap on the CTA
6. Observe that the `cta_tapped` event is logged with the correct properties 

### Screenshots

N/A

---
